fapolicy_analyzer/ui/faprofiler.py

```diff
@@ -0,0 +1,291 @@
+# Copyright Concurrent Technologies Corporation 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import pwd
+import subprocess
+import time
+from .actions import NotificationType, add_notification
+from .fapd_manager import FapdMode
+from .store import dispatch
+from datetime import datetime as DT
+from enum import Enum
+
+import fapolicy_analyzer.ui.strings as strings
+
+
+class ProfSessionStatus(Enum):
+    QUEUED = 0
+    INPROGRESS = 1
+    COMPLETED = 2
+
+
+class FaProfSession:
+    def _demote(enable, user_uid, user_gid):
+        def result():
+            os.setgid(user_gid)
+            os.setuid(user_uid)
+        if enable:
+            return result
+        else:
+            return None
```

Comment:
```suggestion
        return result if enable else None
```

Reply:
You mentioned this one before somewhere, and I missed it... Will do.

Reply:
Done